azure-pipelines-pr-validation.yml

```diff
@@ -0,0 +1,202 @@
+resources:
+- repo: self
+  clean: true
+
+# Variables defined in yml cannot be overridden at queue time instead overrideable variables must be defined in the web gui.
+# Commenting out until AzDO supports something like the runtime parameters outlined here: https://github.com/Microsoft/azure-pipelines-yaml/pull/129
+#variables:
+#  SignType: test
+#  SkipTests: true
+#  SkipApplyOptimizationData: false
+#  IbcDrop: 'default'
+#  PRNumber: 'REQUIRED'
+#  CommitSHA: 'REQUIRED'
+
+# The variables `_DotNetArtifactsCategory` and `_DotNetValidationArtifactsCategory` are required for proper publishing of build artifacts. See https://github.com/dotnet/roslyn/pull/38259
+variables:
+  - name: _DotNetArtifactsCategory
+    value: .NETCore
+  - name: _DotNetValidationArtifactsCategory
+    value: .NETCoreValidation
+  - group: DotNet-Roslyn-SDLValidation-Params
+
+  # To retrieve OptProf data we need to authenticate to the VS drop storage.
+  # If the pipeline is running in DevDiv, the account has access to the VS drop storage.
+  # Get $AccessToken-dotnet-build-bot-public-repo from DotNet-GitHub-Versions-Repo-Write
+  - group: DotNet-GitHub-Versions-Repo-Write
+  - name: _DevDivDropAccessToken
+    value: $(System.AccessToken)
+
+stages:
+- stage: build
+  displayName: Build and Test
+
+  jobs:
+
+  - job: OfficialTestBuild
+    displayName: Official Test Build
+    timeoutInMinutes: 360
+    # Conditionally set build pool so we can share this YAML when building with different pipeline
+    pool:
+      name: VSEngSS-MicroBuild2017
+      demands:
+      - msbuild
+      - visualstudio
+      - DotNetFramework
+
+    steps:
+    - powershell: Write-Host "##vso[task.setvariable variable=SourceBranchName]$('$(Build.SourceBranch)'.Substring('refs/heads/'.Length))"
+      displayName: Setting SourceBranchName variable
+      condition: succeeded()
+
+    - task: tagBuildOrRelease@0
+      displayName: Tag PR validation build
+      inputs:
+        type: 'Build'
+        tags: |
+          PRValidationBuild
+          PRNumber:$(PRNumber)
```

Comment:
Should we also include a `CommitSHA:$(CommitSHA)` tag, for example, since it appears that the PR commit SHA is not related to the commit ID associated with the build.

Reply:
I like this suggestion